maven_iuvs/graphics/line_fit_plot.py

```diff
@@ -3,36 +3,61 @@
 import numpy as np
 import matplotlib as mpl
 import matplotlib.pyplot as plt
+import matplotlib.transforms as transforms
 
+from maven_iuvs.fits_processing import get_pix_range
+from maven_iuvs.constants import nan_color
 
-def detector_image(myfits, integration=0,
-                   fig=None, ax=None,
-                   norm=None, cmap=109,
-                   scale="linear",
-                   arange=None, prange=None):
-    """Makes a density plot of detector counts for the input FITS file and
+# from maven_iuvs.echelle import slit_start, slit_end
+
+
+def detector_image(myfits, image_to_plot=None, integration=0, draw_slit_lines=False, slit_lines=[346, 535],
+                   fig=None, ax=None, norm=None, cmap=109, titletext="", fontsizes="medium",
+                   scale="linear", print_scale_type=True, show_colorbar=True, show_cbar_lbl=True, labels_off=False, 
+                   arange=None, prange=None, force_vmin=None, force_vmax=None):
+    """Makes a density plot of detector DN for the input FITS file and
     specified integration.
 
     Parameters
     ----------
     myfits : IUVSFITS or HDUList
-        IUVS FITS file to be plotted.
+             IUVS FITS file to be plotted.
+    image_to_plot : optional data override to use when plotting.
```

Comment:
this probably indicates a need to split routines up: one for plotting an input array, one to extract the data from a fits file and pass is to the plotting routine. Recommend refactoring.

Reply:
Fixed by 19ff4ea22866d14108ef61c8ecf5b25a8d7f81d7